### PR TITLE
sanitize project name

### PR DESCRIPTION
**Pull request by cdebourcy**

### CODE REVIEW

File: app/controllers/phylo_trees_controller.rb [hunk 1]
```diff
@@ -51,9 +51,8 @@ def index
                  name: taxon_lineage.name }
     end
 
-    @phylo_trees = @phylo_trees.as_json(include: { user: { only: [:id, :name] } })
-
```
Comment (cdebourcy) [inline app/controllers/phylo_trees_controller.rb]: This change is unrelated to the main diff, sorry for mixing in. It completes a change I had made previously (user data is filled in a few lines below, so no need for inefficient "include: user"). I had neglected to remove the now unnecessary code.

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -1,7 +1,21 @@
 require 'csv'
 require 'json'
+require 'open3'
 
 module ApplicationHelper
+  def sanitize(user_input_text)
+    # accept alpanumeric characters only
+    user_input_text.gsub(/[^A-Za-z0-9]/, '-')
```
Comment (jshoe) [inline app/helpers/application_helper.rb]: for the regex, we still want to accept spaces in the project names right? I think we can include space. As long as the params are separated Open3 will handle the spaces apparently:

```
irb(main):010:0> stdin, stdout, stderr = popen3("aws", "s3", "ls", "s3://idseq-samples-staging && echo 'HELLO'")
=> [#<IO:fd 18>, #<IO:fd 20>, #<IO:fd 22>, #<Process::Waiter:0x00007fa44f14e4c0 sleep>]
irb(main):011:0> puts stdout.read

=> nil
irb(main):012:0> stderr.read
=> "\nParameter validation failed:\nInvalid bucket name \"idseq-samples-staging && echo 'HELLO'\": Bucket name must match the regex \"^[a-zA-Z0-9.\\-_]{1,255}$\"\n"
irb(main):013:0>
```

Comment (cdebourcy) [reply app/helpers/application_helper.rb]: That's true. I will make the regex more permissive

Comment (cdebourcy) [reply app/helpers/application_helper.rb]: thanks

Comment (jshoe) [reply app/helpers/application_helper.rb]: "_" and "-" too probably

File: app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -254,6 +255,7 @@ def set_project
 
   # Never trust parameters from the scary internet, only allow the white list through.
   def project_params
+    params[:name] = sanitize(params[:name])
```
Comment (jshoe) [inline app/controllers/projects_controller.rb]: :name is nested right?

Comment (cdebourcy) [reply app/controllers/projects_controller.rb]: yes. thanks

Comment (cdebourcy) [reply app/controllers/projects_controller.rb]: moved sanitize to after the require/permit, at which point name is no longer nested

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -1,7 +1,21 @@
 require 'csv'
 require 'json'
+require 'open3'
 
 module ApplicationHelper
+  def sanitize(user_input_text)
+    # accept alpanumeric characters only
```
Comment (jshoe) [inline app/helpers/application_helper.rb]: nit: alphanumeric. or 'limited' or something

Comment (cdebourcy) [reply app/helpers/application_helper.rb]: ok fixed

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -1,7 +1,21 @@
...
+    user_input_text.gsub(/[^A-Za-z0-9_\- ]/, '-')
+  end
+
+  def safe_s3_rm(s3_path)
```
Comment (tfrcarvalho) [inline app/helpers/application_helper.rb]: Do you mind putting these two commands in AWS dedicated helper ? 
I would like to centralize every AWS access into AWS access layer.

File: app/helpers/application_helper.rb (hunk 4 above)
Comment (tfrcarvalho) [inline app/helpers/application_helper.rb]: typo: alpanumeric -> alphanumeric

Maybe change to "accept alphanumeric characters and symbols `_` and `-`"

File: app/helpers/application_helper.rb (hunk 5 above)
Comment (cdebourcy) [reply app/helpers/application_helper.rb]: agreed
